haskell-style.md

```diff
@@ -660,3 +660,329 @@ module Driver
+  -- est laborum.
+  theFunction
+
+  -- | The docs
+  --
+  -- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  -- tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat.
+  -- Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  -- dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  -- proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  theFunction
+  ```
+
+  Good
+
+  ```hs
+  -- | The docs
+  --
+  -- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  -- tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat.
+  --
+  -- Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  -- dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  -- proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  --
+  theFunction
+  ```
+
+- Lists receive a hanging indent
+
+  Bad
+
+  ```hs
+  -- | The docs
+  --
+  -- 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  -- tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  -- veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  -- commodo consequat.
+  -- 2. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  -- dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  -- proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  --
+  theFunction
+  ```
+
+  Good
+
+  ```hs
+  -- | The docs
+  --
+  -- 1. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+  --    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+  --    veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+  --    commodo consequat.
+  -- 2. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+  --    dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+  --    proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+  --
+  theFunction
+  ```
+
+### Module organization
+
+- Organize your exports by logical groups or progressive disclosure and use
+  section headings
+
+  Bad
+
+  ```hs
+  module Foo
+    ( getFoo
+    , updateBar
+    , internalDeleteFoo
+    , deleteFoo
+    )
+  ```
+
+  Good
+
+  ```hs
+  module Foo
+    (
+    -- * Foo
+      getFoo
+    , deleteFoo
+
+    -- * Bar
+    , updateBar
+
+    -- * Internal, exported for testing
+    , internalDeleteFoo
+    )
+  ```
+
+- Consider adding section documentation
+
+  **NOTE**: Summary/Body rules apply!
+
+  ```hs
+  module Foo
+    (
+    -- * Foo
+    -- | Operates on Foos
```

Comment:
Same comment I had about indenting 2 spaces in records. Having the comment start 2 spaces less indented than the thing we are commenting feels a bit funny.

Reply:
> Having the comment start 2 spaces less indented than the thing we are commenting feels a bit funny.

That's just an awkward thing that happens when documenting the first export in a module list (and it doesn't happen with attributes or arguments because they are documented trailing-style). I've gotten used to it, and I wouldn't want to add an indentation to the Guide overall because of this first element quirk.

That's the other perspective anyway, I'd love to hear other voices from the team.

Reply:
Ah I wasn't talking about the first comma missing. I was talking about not aligning to the comma, but the identifier. In fact this style doesn't suffer from the awkwardness of the first comma missing

```haskell
module Foo
  (
    -- * Foo
    -- | Operates on Foos
    getFoo
  , deleteFo

    -- * Bar
    -- | Operates on Bars
  , updateBar
  ...
  )

Reply:
Both styles look fine to me, no strong opinion